cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java

```diff
@@ -265,7 +265,7 @@ private void setConf(CConfiguration conf) {
   }
 
   public CConfiguration getConf() {
-    CConfiguration conf = CConfiguration.create();
+    CConfiguration conf = CConfiguration.createEmpty();
```

Comment:
There are many places we do similar thing by calling `cConf.clear()` first (search for usage of clear(), you will see a quite a number). It's better to unify to do it in one way. I propose to do it like this:

```java
CConfiguration cConf = CConfiguration.create(resource);

...
// Two new factory method to create with the give resources only
public static CConfiguration create(URI resource, URI...moreResources);
public static CConfiguration create(InputStream resource);
```